opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -247,12 +248,22 @@ CpGrid::scatterGrid(EdgeWeightMethod method, const std::vector<cpgrid::OpmWellTy
 
         distributed_data_->distributeGlobalGrid(*this,*this->current_view_data_, cell_part);
         int num_cells = distributed_data_->cell_to_face_.size();
-        std::ostringstream message;
-        message << "After loadbalancing process " << my_num << " has " << num_cells << " cells.";
-        if (num_cells == 0) {
-            throw std::runtime_error(message.str() + " Aborting.");
-        } else {
-            std::cout << message.str() << "\n";
+
+        cc_num_cells[cc.rank()] = num_cells;
+        cc.sum(cc_num_cells.data(),cc.size());
+
+        if (cc.rank() == 0) {
+            std::string str = "\nAfter loadbalancing on " + std::to_string(cc.size()) + " processes we have\nrank: num cells\n----------------\n";
+            for (int i = 0; i < cc.size(); ++i) {
+                str += std::to_string(i) + ": " + std::to_string(cc_num_cells[i]) + "\n";
+            }
+            str += "----------------\n";
+            Opm::OpmLog::info(str);
+        }
+        for(const auto& cc_num_cells: nc) {
```

Comment:
You must put nc before cc_num_cells.

Also some nitpicks: space after for, space around colon, and move the definition of cc_num_cells to line 251 instead of 100 lines above.

Reply:
done